### ISSUE TITLE
[Held requests] Hold Request education modal reappears after relogin

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#37012]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Found when validating PR :** https://github.com/Expensify/App/pull/33897

**Version Number:** v1.4.43-6
**Reproducible in staging?:** y
**Reproducible in production?:** n
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** applause-Internal Team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com.
2. Go to workspace chat.
3. Create a manual request.
4. Go to request details page.
5. Click 3-dot menu > Hold request.
6. Add a reason and submit it.
7. Log out.
8. Log in.
9. Go to request details page.
10. Click Got it on the Hold request education modal.
11. Repeat Step 6-8.

## Expected Result:
Hold Request education modal will not appear again after relogin.

## Actual Result:
Hold Request education modal reappears after relogin.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/95209e25-7cf9-4edb-b8c0-fefd68a78bbb

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0135c085e6de6bb2c1</li>
        <li>Upwork Job ID: 1760290228470611968</li>
        <li>Last Price Increase: 2024-02-21</li>
    </ul>
</details>

**Comment by izarutskaya** [comment]
We think that this bug might be related to #vip-bills
CC @davidcardoza

**Comment by rlinoz** [comment]
I don't think this is a blocker, but I am also unsure about the expected behavior here, if this should never show to the same user we may need some backend fix.

CC'ing @robertjchen as you might know the expected behavior

**Comment by robertjchen** [comment]
Yeah, definitely not a blocker. I guess we can commit this to the user's NVP to prevent future display 🤔 Let me check on the proper approach here.

**Comment by cdOut** [comment]
Hello, I'm Tymoteusz Kałuzieński from the expert agency Software Mansion, and I'd like to work on this issue.

I implemented the hold interstitial and it already uses local NVP variables to store whether or not the modal has been shown to the user, at first glance it looks like there's a problem with properly saving and reading Onyx state. I'll investigate and come up with a fix.

**Comment by cdOut** [comment]
After investigating this issue, it turned out we didn't think about this case when considering storage for saving whether a user has seen the interstitial or not. Currently we're using **Onyx storage for it, which gets cleared out on logging out**. I've messaged @robertjchen on slack about it, we'll discuss changing the way we save the value and I'll implement it after.

**Comment by robertjchen** [comment]
Thanks- let me know if there's anything I can assist with from the backend 👍

**Comment by robertjchen** [comment]
Chatted 1:1, working on the necessary backend changes

**Comment by robertjchen** [comment]
@cdOut I'm creating a new API command called `DismissFeatureIntro` which will just take one `string` parameter `feature` 
- Each educational interstitial will have a `feature` name (up to your choosing)
- When the interstitial is acknowledged/dismissed, we'll call the `DismissFeatureIntro` API command with the `feature` name.
- On the backend, this command will simply add the feature name to a list in a NVP called `NVP_INTRODUCED_FEATURES` : `introducedFeatures`.
- This will be merged into Onyx and be in the form of `feature1,feature2` which the frontend can check to determine if an interstitial should be shown.

Hopefully that's enough to get started on the frontend portion, let me know if the design is ok 👍

**Comment by cdOut** [comment]
I’ll create a draft PR today.

**Comment by cdOut** [comment]
No update from me, waiting for backend solutions to be implemented and confirmed.

**Comment by robertjchen** [comment]
blocked on internal issue to revamp how NVPs are sync'd on frontend/backend (which will store this setting). https://github.com/Expensify/Expensify/issues/364560

**Comment by trjExpensify** [comment]
@robertjchen I don't think this issue needs to be held anymore, can you confirm and remove the hold?

**Comment by robertjchen** [comment]
Yep, confirmed we are no longer blocked on this.

@cdOut I see the PR is in draft, but it seems ready for review- can you please confirm? 🙏

**Comment by cdOut** [comment]
I'll merge with main, write the tests and videos and push it through review. Thanks for the reminder!

**Comment by cdOut** [comment]
I'm currently trying to work out issues that appeared after merging with main, I should have it in review by tomorrow.

**Comment by cdOut** [comment]
Hi, I've came across some problems during main merge, will update it and complete the PR soon.

**Comment by robertjchen** [comment]
bump @cdOut 🙏

**Comment by cdOut** [comment]
I'm on it, sorry for the delays 🙇

**Comment by robertjchen** [comment]
Discussion/Clarification ongoing in PR, we're close!

**Comment by robertjchen** [comment]
Final discussion/tests 🙏

**Comment by robertjchen** [comment]
Hoping to get this wrapped up this week, we're close!

**Comment by robertjchen** [comment]
Backend fix under review and to be deployed!
